core/src/main/java/com/klaviyo/coresdk/ConfigFileUtils.kt

```diff
@@ -0,0 +1,45 @@
+package com.klaviyo.coresdk
+
+import java.io.File
+import java.util.*
+
+internal object ConfigFileUtils {
+    private const val CONFIG_FILE_NAME = "klaviyo-sdk-config"
+
+    private const val UUID_KEY = "uuid"
+
+    fun readOrCreateUUID(): String {
+        val file = File("${KlaviyoConfig.applicationContext.filesDir}/$CONFIG_FILE_NAME")
```

Comment:
I went with internal storage for the UUID since it is then stored alongside the parent application. When the parent application is uninstalled, or when app data is cleared, this file will be deleted. This gives us pretty much consistent anonymous IDs for each install of the application, which works as a basic initial solution to our problem.

Reply:
I went for internal storage over shared preferences since its more private to the parent application and I'm not particularly fond of the idea of this just being publicly accessible and changeable information. With internal storage, it is at least protected from access via other apps.
To avoid this logic, the alternative is to utilize Firebase IIDs, which is a lib that will make generating the UUIDs pretty straightforward for this use case. Since we plan to change how we identify anonymous IDs in the future I decided not to utilize this dependency and instead do a manual solution that is more generic.

Reply:
Nice sounds like a solid choice!